source/common/network/address_impl.cc

```diff
@@ -230,8 +255,8 @@ Api::SysCallIntResult Ipv4Instance::bind(os_fd_t fd) const {
 }
 
 Api::SysCallIntResult Ipv4Instance::connect(os_fd_t fd) const {
-  const int rc = ::connect(fd, sockAddr(), sockAddrLen());
-  return {rc, errno};
+  return Api::OsSysCallsSingleton::get().connect(
+      fd, reinterpret_cast<const sockaddr*>(&ip_.ipv4_.address_), sizeof(ip_.ipv4_.address_));
```

Comment:
Can we use sockAddr() and sockAddrLen() here and below?

Reply:
changed this, resolving